### PR TITLE
Break out certificate playbooks

### PR DESCRIPTION
**Pull request by abutcher**
Break out certificate role dependencies and move them into their own playbooks.

### CODE REVIEW

**Comment by ingvagabund** [comment]
My first rpm based deployment errored with:
```yaml
TASK [openshift_node_certificates : Copy OpenShift CA to system CA trust] ********************************************************************************************************************************************************************
changed: [10.8.173.206] => (item={'cert': '/etc/origin/node/ca.crt', 'id': 'openshift'})

RUNNING HANDLER [openshift_node_certificates : update ca trust] ******************************************************************************************************************************************************************************
changed: [10.8.173.206]

RUNNING HANDLER [openshift_node_certificates : restart docker after updating ca trust] *******************************************************************************************************************************************************
FAILED - RETRYING: restart docker after updating ca trust (3 retries left).
FAILED - RETRYING: restart docker after updating ca trust (2 retries left).
FAILED - RETRYING: restart docker after updating ca trust (1 retries left).
fatal: [10.8.173.206]: FAILED! => {"attempts": 3, "changed": false, "failed": true, "msg": "Could not find the requested service docker: host"}
```

Maybe a glitch, re-runing again.

**Comment by ingvagabund** [comment]
Running `playbooks/byo/openshift-cluster/redeploy-certificates.yml` failed with:
```yaml
TASK [Redeploy router] ***********************************************************************************************************************************************************************************************************************
fatal: [10.8.173.202]: FAILED! => {"changed": true, "cmd": ["oc", "deploy", "dc/router", "--latest", "--config=/tmp/openshift-ansible-YMpdUg/admin.kubeconfig", "-n", "default"], "delta": "0:00:00.153343", "end": "2017-09-29 09:29:53.459563", "failed": true, "rc": 1, "start": "2017-09-29 09:29:53.306220", "stderr": "Command \"deploy\" is deprecated, Use the `rollout latest` and `rollout cancel` commands instead.\nFlag --latest has been deprecated, use 'oc rollout latest' instead\nerror: #1 is already in progress (Pending).\nOptionally, you can cancel this deployment using 'oc rollout cancel dc/router'.", "stderr_lines": ["Command \"deploy\" is deprecated, Use the `rollout latest` and `rollout cancel` commands instead.", "Flag --latest has been deprecated, use 'oc rollout latest' instead", "error: #1 is already in progress (Pending).", "Optionally, you can cancel this deployment using 'oc rollout cancel dc/router'."], "stdout": "", "stdout_lines": []}
	to retry, use: --limit @/home/jchaloup/Projects/src/github.com/ingvagabund/openshift-ansible/playbooks/byo/openshift-cluster/redeploy-certificates.retry
```

File: roles/etcd/tasks/certificates/remove_generated_certificates.yml [hunk 1]
```diff
@@ -1,5 +1,11 @@
...
+- name: Determine if generated etcd certificates exist
+  stat:
+    path: "{{ etcd_conf_dir }}/generated_certs"
+  register: etcd_generated_certs_dir_stat
+
 - name: Remove generated etcd certificates
   file:
     path: "{{ etcd_conf_dir }}/generated_certs"
     state: absent
+  when: etcd_generated_certs_dir_stat.stat.exists | bool
```
Comment (ingvagabund) [inline roles/etcd/tasks/certificates/remove_generated_certificates.yml]: It the condition necessary? The `absent` should by idempotent. If the directory exists, it get deleted, if it does not exist, nothing will happen. For that reason, I removed this piece of code intentionally.

File: playbooks/common/openshift-node/config.yml [hunk 2]
```diff
@@ -69,7 +68,6 @@
   roles:
   - role: os_firewall
   - role: openshift_node
-    openshift_ca_host: "{{ groups.oo_first_master.0 }}"
```
Comment (ingvagabund) [inline playbooks/common/openshift-node/config.yml]: Cool, `openshift_ca_host` needed only when the `openshift_node_certificates` is run.

File: playbooks/common/openshift-master/config.yml [hunk 3]
```diff
@@ -204,15 +204,6 @@
...
   - role: openshift_hosted_facts
-  - role: openshift_master_certificates
-  - role: openshift_etcd_facts
-  - role: openshift_etcd_client_certificates
-    etcd_cert_subdir: "openshift-master-{{ openshift.common.hostname }}"
-    etcd_cert_config_dir: "{{ openshift.common.config_base }}/master"
-    etcd_cert_prefix: "master.etcd-"
-    r_etcd_common_etcd_runtime: "{{ openshift.common.etcd_runtime }}"
-    etcd_ca_host: "{{ groups.oo_etcd_to_config.0 }}"
-    when: groups.oo_etcd_to_config | default([]) | length != 0
```
Comment (ingvagabund) [inline playbooks/common/openshift-master/config.yml]: +1 to moving the deps out of the `openshift-master/config.yml`.

With the certificates roles removed, does it mean that when i run `playbooks/byo/openshift-master/config.yml` the certificates will not get deployed? Or is the certs deployment just missing in the byo equivalent?

File: playbooks/common/openshift-cluster/redeploy-certificates/etcd-ca.yml [hunk 4]
```diff
@@ -72,7 +59,7 @@
       name: etcd
       tasks_from: retrieve_ca_certificates
     vars:
-      etcd_sync_cert_dir: hostvars['localhost'].g_etcd_mktemp.stdout
+      etcd_sync_cert_dir: "{{ hostvars['localhost'].g_etcd_mktemp.stdout }}"
```
Comment (ingvagabund) [inline playbooks/common/openshift-cluster/redeploy-certificates/etcd-ca.yml]: +1

File: roles/etcd/tasks/certificates/remove_generated_certificates.yml (hunk 1 above)
Comment (abutcher) [reply roles/etcd/tasks/certificates/remove_generated_certificates.yml]: You're right, I'll take this out.

File: playbooks/common/openshift-master/config.yml (hunk 3 above)
Comment (abutcher) [reply playbooks/common/openshift-master/config.yml]: Right, the certificate playbooks are currently not included in the `config.yml` playbooks and are included one level up.

**Comment by abutcher** [comment]
Note that the router failure above `error: #1 is already in progress (Pending)` is a known problem with redeploying deploymentConfigs and that the router redeploy is broken for the default install case in a separate bug. Fixed by https://github.com/openshift/openshift-ansible/pull/5449.

**Pull request merged**